### PR TITLE
feat: add `sagemaker-studio` module 

### PR DESCRIPTION
**Pull request by kukushking**
*Description of changes:*
- Added SageMaker Studio MLOps module (moving from IDF repo to dedicated MLOps repo)
- Upgraded CDK version to the latest (2.128.0)
- Added architecture diagram to the docs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dgraeber — CHANGES_REQUESTED** [review]
Missing CHANGELOG entry

File: manifests/sagemaker-studio-modules.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: studio
-path: git::https://github.com/awslabs/idf-modules.git//modules/ml/sagemaker-studio?ref=release/1.3.0&depth=1
+path: modules/sagemaker/sagemaker-studio
```
[the remark this answers is not in the document]
Comment (kukushking) [reply manifests/sagemaker-studio-modules.yaml:2]: Thanks, yes, will lock all module versions after the release is cut

File: modules/sagemaker/sagemaker-studio/README.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  - name: lead_data_science_users
+    value:
+    - lead-data-scientist
+```
+
+3. Default SageMaker Project Templates are also enabled on the account on the targeted region using a custom resource; the custom resource uses a lambda function, `functions/sm_studio/enable_sm_projects`, to make necessary SDK calls to both Amazon Service Catalog and Amazon SageMaker.
+
+## Inputs and outputs:
+### Required inputs:
+  - `VPC_ID`
```
Comment (srinivasreddych) [inline modules/sagemaker/sagemaker-studio/README.md:40]: description of few fields are missing

File: modules/sagemaker/sagemaker-studio/README.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+This is an AWS CDK project written in Python 3.8. Here's what you need to have on your workstation before you can deploy this project. It is preferred to use a linux OS to be able to run all cli commands and avoid path issues.
+
+* [Node.js](https://nodejs.org/)
+* [Python3.8](https://www.python.org/downloads/release/python-380/) or [Miniconda](https://docs.conda.io/en/latest/miniconda.html)
+* [AWS CDK v2](https://aws.amazon.com/cdk/)
+* [AWS CLI](https://aws.amazon.com/cli/)
+* [Docker](https://docs.docker.com/desktop/)
+
+### Module Structure
```
Comment (srinivasreddych) [inline modules/sagemaker/sagemaker-studio/README.md:86]: Best explanation seen so far :)

**Review by srinivasreddych — APPROVED** [review]

**Review by srinivasreddych — APPROVED** [review]

File: modules/sagemaker/sagemaker-studio/README.md (hunk 2 above)
Comment (kukushking) [reply modules/sagemaker/sagemaker-studio/README.md]: Oops... thanks!

**Review by dgraeber — APPROVED** [review]

**Pull request merged**